### PR TITLE
Simulation drivetrain

### PR DESCRIPTION
**Pull request by m10653**
Added Simulation support to drive train and  added build information to network tables and logs

### CODE REVIEW

**Comment by m10653** [comment]
@tjamesw123 Please Check the Navx.json dependency change. Want to make sure I did not break the navx. (I just updated it to the latest version from the navx site)

File: vendordeps/NavX.json [hunk 1]
```diff
@@ -1,35 +1,36 @@
...
     "name": "KauaiLabs_navX_FRC",
-    "version": "2023.1.0",
+    "version": "2023.0.3",
     "uuid": "cb311d09-36e9-4143-a032-55bb2b94443b",
     "mavenUrls": [
-        "https://raw.githubusercontent.com/rzblue/navx-frc/maven/maven/release/"
+        "https://dev.studica.com/maven/release/2023/"
     ],
-    "jsonUrl": "https://raw.githubusercontent.com/rzblue/navx-frc/maven/navx_frc.json",
+    "jsonUrl": "https://dev.studica.com/releases/2023/NavX.json",
```
Comment (tjamesw123) [inline vendordeps/NavX.json:9]: Did you find a different third party to get NavX code from?

**Review by tjamesw123 — APPROVED** [review]
As long as the new third party code you have linked works I think it's fine

**Comment by tjamesw123** [comment]
The NavX code from the site is most likely still broken but we can try however most likely we will have to revert that

File: vendordeps/NavX.json (hunk 1 above)
Comment (m10653) [reply vendordeps/NavX.json:9]: I used the json from the navx website https://pdocs.kauailabs.com/navx-mxp/software/roborio-libraries/java/
https://dev.studica.com/releases/2023/NavX.json

The version that was imported before appeared to be out of date and was causing crashes when running

**Pull request merged**